large_image/tilesource/resample.py

```diff
@@ -0,0 +1,124 @@
+from enum import Enum
+from typing import Dict
+
+import numpy as np
+from PIL import Image
+
+
+class ResampleMethod(Enum):
+    PIL_NEAREST = Image.Resampling.NEAREST  # 0
+    PIL_LANCZOS = Image.Resampling.LANCZOS  # 1
+    PIL_BILINEAR = Image.Resampling.BILINEAR  # 2
+    PIL_BICUBIC = Image.Resampling.BICUBIC  # 3
+    PIL_BOX = Image.Resampling.BOX  # 4
+    PIL_HAMMING = Image.Resampling.HAMMING  # 5
+    PIL_MAX_ENUM = 5
+    NP_MEAN = 6
+    NP_MEDIAN = 7
+    NP_MODE = 8
+    NP_MAX = 9
+    NP_MIN = 10
+    NP_NEAREST = 11
+    NP_MAX_CROSSBAND = 12
+    NP_MIN_CROSSBAND = 13
+
+
+def pilResize(
+    tile: np.ndarray,
+    new_shape: Dict,
+    resample_method: ResampleMethod,
+) -> np.ndarray:
+    # Only NEAREST works for 16 bit images
+    img = Image.fromarray(tile)
+    resized_img = img.resize(
+        (new_shape['width'], new_shape['height']),
+        resample=resample_method.value,
+    )
+    result = np.array(resized_img).astype(tile.dtype)
+    return result
+
+
+def numpyResize(
+    tile: np.ndarray,
+    new_shape: Dict,
+    resample_method: ResampleMethod,
+) -> np.ndarray:
+    if resample_method == ResampleMethod.NP_NEAREST:
+        return tile[::2, ::2]
+    else:
+        pixel_selection = None
+        subarrays = np.asarray(
+            [
```

Comment:
What happens if the tile's shape is not a multiple of 2?

Reply:
We probably could replicate the last row and/or column in those instances.

Reply:
Added in [dbcb661](https://github.com/girder/large_image/pull/1490/commits/dbcb661c038b87058b7d2975d5952b43b38b46d3)